x-pack/plugin/rollup/src/main/java/org/elasticsearch/xpack/rollup/v2/FieldTypeHelper.java

```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.rollup.v2;
+
+import java.util.Map;
+
+public interface FieldTypeHelper {
```

Comment:
This interface exists because I dont want to leak the actual class implementation when I do this:

```
public FieldTypeHelper timeseriesHelper(final String timestampField) throws IOException {
            final MapperService mapperService = indicesService.createIndexMapperServiceForValidation(indexMetadata);
            final CompressedXContent sourceIndexCompressedXContent = new CompressedXContent(indexMapping);
            mapperService.merge(MapperService.SINGLE_MAPPING_NAME, sourceIndexCompressedXContent, MapperService.MergeReason.INDEX_TEMPLATE);
            return new TimeseriesFieldTypeHelper(mapperService, timestampField);
        }
```

So it is there to avoid the client knowing which actual class implementation we are using.